### ISSUE TITLE
[java] Wrong name for inner classes in violations

### ISSUE THREAD

**Issue by andipabst** [pmd/pmd#2105]
**Affects PMD Version:** 
6.19.0

**Rule:**
any

**Description:**
When a Java class has inner classes, their classname is not correctly reported when looking at the class itself. This is not an issue for code inside a class, but the inner class declaration itself.

**Code Sample demonstrating the issue:**
```java
class Foo { // classname: Foo -> correct
    class Bar { // classname Foo -> wrong, should be Foo$Bar
    } 
}
```

**Running PMD through:** *[Java API | any]*
